src/core/lib/transport/timeout_encoding.cc

```diff
@@ -23,83 +23,214 @@
 #include <stdio.h>
 #include <string.h>
 
+#include <grpc/support/log.h>
+
 #include "src/core/lib/gpr/string.h"
 
-static int64_t round_up(int64_t x, int64_t divisor) {
-  return (x / divisor + (x % divisor != 0)) * divisor;
+namespace grpc_core {
+
+namespace {
+
+int64_t DivideRoundingUp(int64_t dividend, int64_t divisor) {
+  return (dividend + divisor - 1) / divisor;
 }
 
-/* round an integer up to the next value with three significant figures */
-static int64_t round_up_to_three_sig_figs(int64_t x) {
-  if (x < 1000) return x;
-  if (x < 10000) return round_up(x, 10);
-  if (x < 100000) return round_up(x, 100);
-  if (x < 1000000) return round_up(x, 1000);
-  if (x < 10000000) return round_up(x, 10000);
-  if (x < 100000000) return round_up(x, 100000);
-  if (x < 1000000000) return round_up(x, 1000000);
-  return round_up(x, 10000000);
+constexpr int64_t kSecondsPerMinute = 60;
+constexpr int64_t kMinutesPerHour = 60;
+constexpr int64_t kSecondsPerHour = kSecondsPerMinute * kMinutesPerHour;
+constexpr int64_t kMaxHours = 27000;
+
+bool IsAllWhitespace(const uint8_t* p, const uint8_t* end) {
```

Comment:
nit: `IsAllSpace` since it's not covering all whitespaces?

Reply:
done!